src/Symfony/Component/Form/Extension/Core/Type/EnumType.php

```diff
@@ -31,8 +31,21 @@ public function configureOptions(OptionsResolver $resolver): void
             ->setAllowedValues('class', enum_exists(...))
             ->setDefault('choices', static fn (Options $options): array => $options['class']::cases())
             ->setDefault('choice_label', static function (Options $options) {
-                if (\is_array($options['choices']) && !array_is_list($options['choices'])) {
-                    return null;
+                $choices = $options['choices'];
+
+                if (\is_array($choices) && !array_is_list($choices)) {
```

Comment:
What happens if you have a mix of grouped and non-grouped choices, for example:

```php
'choices' => [
    'Group 1' => [
        'Custom Yes' => Answer::Yes,
        'Custom No' => Answer::No,
    ],
    'Custom 42' => Answer::FortyTwo,
],
```

Also, what if you have a mix of labeled and unlabeled choices:

```php
'choices' => [
    Answer::Yes,
    Answer::No,
    'Custom 42' => Answer::FortyTwo,
],
```

Reply:
hi @HypeMC, thank you for your feedback!

I have improved the logic and added tests for grouped and custom-labeled Enum choices.

For hybrid cases (mixing grouped, non-grouped, labeled, and non-labeled choices), the expected behavior isn’t so clear.

Should we make it flexible and auto-detect labels, or do you prefer a strict approach (numeric indexes for non-labeled, labels only if set)?
What do you think is best for EnumType in Symfony?